src/oc_erchef/apps/chef_index/src/chef_solr.erl

```diff
@@ -24,17 +24,80 @@
 -module(chef_solr).
 
 -export([
+         %% Static Data Accessors
+         search_provider/0,
+         id_field/0,
+         database_field/0,
+         kv_sep/0,
+         type_field/0,
+         update_url/0,
+         %% Query Building
          add_org_guid_to_query/2,
-         delete_search_db/1,
-         delete_search_db_by_type/2,
          make_query_from_params/4,
-         ping/0,
+         transform_query_all/1,
+         transform_query_safe/1,
+         transform_query_term/1,
+         transform_query_phrase/1,
+         %% Document Buidling
+         transform_data/1,
+         %% Operations
          search/1,
-         solr_commit/0
+         update/1,
+         commit/0,
+         ping/0,
+         delete_search_db/1,
+         delete_search_db_by_type/2,
+         %% Helpers used by providers
+         db_from_orgid/1
         ]).
 
 -include("chef_solr.hrl").
+-define(PROVIDER_FUNC(FuncName),
+        begin
+            Mod = list_to_atom(atom_to_list(search_provider()) ++ "_provider"),
```

Comment:
I think we talked about having the config render the actual provider name? That would let these look like: 
```
-define(PROVIDER_FUNC(FuncName) (search_provider()):FuncName() ).
```
and 
```
-define(PROVIDER_FUNC(FuncName,Args) (search_provider()):FuncName(Args) ).
```